test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemTest.scala

```diff
@@ -168,28 +168,14 @@ class SystemTest {
     val list2 = List(1, 3, 5)
     assertEquals(list2, list1)
     assertEquals(list2.hashCode(), list1.hashCode())
-    assertNotEquals(System.identityHashCode(list1), System.identityHashCode(list2))
+    if (!executingInJVM)
```

Comment:
Because the JVM does not guarantee that two objects allocated one after the other have different id hash codes. (We don't guarantee that either, but our implementation happens to always satisfy that, so we can test it that way for JS only.)